libcnb-data/src/exec_d.rs

```diff
@@ -1,5 +1,111 @@
+use crate::newtypes::libcnb_newtype;
 use serde::Serialize;
 use std::collections::HashMap;
 
-#[derive(Serialize)]
-pub type ExecD = HashMap<String, String>;
+#[derive(Serialize, Clone)]
+pub struct ExecDProgramOutput(HashMap<ExecDProgramOutputKey, String>);
+
+impl ExecDProgramOutput {
+    #[must_use]
+    pub fn new(map: HashMap<ExecDProgramOutputKey, String>) -> Self {
+        Self(map)
+    }
+}
+
+impl<K: Into<ExecDProgramOutputKey>, V: Into<String>, A: IntoIterator<Item = (K, V)>> From<A>
+    for ExecDProgramOutput
+{
+    fn from(a: A) -> Self {
+        Self(
+            a.into_iter()
+                .map(|(key, value)| (key.into(), value.into()))
+                .collect(),
+        )
+    }
+}
+
+libcnb_newtype!(
+    exec_d,
+    /// Construct a [`ExecDProgramOutputKey`] value at compile time.
+    ///
+    /// Passing a string that is not a valid `ExecDProgramOutputKey` value will yield a compilation
+    /// error.
+    ///
+    /// # Examples:
+    /// ```
+    /// use libcnb_data::exec_d::ExecDProgramOutputKey;
+    /// use libcnb_data::exec_d_program_output_key;
+    ///
+    /// let key: ExecDProgramOutputKey = exec_d_program_output_key!("PATH");
+    /// ```
+    exec_d_program_output_key,
+    /// A key of from exec.d program output
```

Comment:
"key of from" seems like a typo?

Should this rustdoc either link to https://github.com/buildpacks/spec/blob/main/buildpack.md#execd-output-toml or include a bit more detail? From reading the rustdoc at the moment I'd have no idea what this should be used for, without also having read the spec.

Reply:
Typo fixed, link to spec added to `ExecDProgramOutput`.